components/compiler/scanner.cpp

```diff
@@ -282,6 +282,8 @@ namespace Compiler
 
         if (!scanName (name))
             return false;
+        else if(name.empty())
```

Comment:
This shouldn't be necessary in the case you try to avoid (since "error" is not true), but if somehow an incomplete string that didn't pass your check below is empty it could make it possible for the parser to utilize empty strings in more cases than intended. Are you sure it's essential to fix the script?